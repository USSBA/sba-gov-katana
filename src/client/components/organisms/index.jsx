/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
