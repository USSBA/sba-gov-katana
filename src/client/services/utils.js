import querystring from 'querystring'

function getQueryParams() {
  let queryParams = {}
  const { search } = window.location
  if (search && search !== '') {
    queryParams = querystring.decode(search.replace('?', ''))
  }
  return queryParams
}

export { getQueryParams }
