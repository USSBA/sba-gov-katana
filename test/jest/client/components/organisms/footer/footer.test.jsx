import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import { Footer } from 'organisms'
import * as utils from '../../../../../../src/client/services/utils.js'
import { render, cleanup } from 'react-testing-library'
import 'jest-dom/extend-expect'

describe('Front Page Hero', () => {
  let stubGetLanguageOverride
  beforeAll(() => {
    stubGetLanguageOverride = sinon.stub(utils, 'getLanguageOverride')
    stubGetLanguageOverride.returns('en')
  })
  afterAll(() => {
    stubGetLanguageOverride.restore()
  })

  test('should render correctly', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('BackToTopButton', () => {
  it('renders the back-to-top button component', () => {
    const { getByTestId } = render(<Footer />)
    expect(getByTestId('back-to-top-button')).toBeInTheDocument()
  })
})
