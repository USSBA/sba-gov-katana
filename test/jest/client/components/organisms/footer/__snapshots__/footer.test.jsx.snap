// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Front Page Hero should render correctly 1`] = `
<footer
  className="footer"
  id="sba-footer"
>
  <div
    className="footerLinks"
  >
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-0"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-0-title"
            onBlur={[Function]}
          >
            Customer Service
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            About SBA
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Contact SBA
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            SBA en Espa√±ol
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Media and Press Relations
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-4"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            SBA Locations
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-0-5"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            SBA Team
          </a>
        </li>
      </ul>
    </div>
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-1"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-1-title"
            onBlur={[Function]}
          >
            About SBA.gov
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Site Map
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Linking Policy
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Accessibility
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-4"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Disclaimers
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-5"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Social Media
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-6"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Data Store
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-1-7"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Blog
          </a>
        </li>
      </ul>
    </div>
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-2"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-2-title"
            onBlur={[Function]}
          >
            SBA Information
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Freedom of Information Act
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            No Fear Act
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Report Fraud, Waste, and Abuse
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Initiatives
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-4"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Plain Language
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-2-5"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            National Resource Guides (English/Spanish)
          </a>
        </li>
      </ul>
    </div>
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-3"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-3-title"
            onBlur={[Function]}
          >
            SBA Performance
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-3-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Strategic Planning
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-3-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Performance, Budget, and Financing
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-3-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Open Government
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-3-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Policy and Regulations
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-3-4"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Eliminating Fraud, Waste, and Abuse
          </a>
        </li>
      </ul>
    </div>
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-4"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-4-title"
            onBlur={[Function]}
          >
            Oversight
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-4-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Inspector General
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-4-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Advocacy
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-4-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Hearings and Appeals
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-4-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Ombudsman
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            href="https://www.whitehouse.gov"
            id="footer-group-4-4"
            onBlur={[Function]}
          >
            WhiteHouse.gov
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            href="https://www.usa.gov"
            id="footer-group-4-5"
            onBlur={[Function]}
          >
            USA.gov
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            href="https://www.regulations.gov"
            id="footer-group-4-6"
            onBlur={[Function]}
          >
            Regulations.gov
          </a>
        </li>
      </ul>
    </div>
    <div
      className="desktopFooterLinks"
    >
      <ul
        className="pageLinkGroup"
        id="footer-group-5"
      >
        <li>
          <a
            href="javascript:void(0);"
            id="footer-group-5-title"
            onBlur={[Function]}
          >
            Tools and Features
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-5-0"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Online Training
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-5-1"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Find Events
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-5-2"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            Qualify for Government Contracts
          </a>
        </li>
        <li
          className="pageLink "
        >
          <a
            className="link "
            id="footer-group-5-3"
            onBlur={[Function]}
            onClick={[Function]}
            style={Object {}}
            target="_self"
          >
            SBA Videos
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="tabletFooterLinks"
  >
    <ul
      className="pageLinkGroup"
      id="tablet-footer-group-1"
    >
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-1-0"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          What We Do
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-1-1"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA Performance
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-1-2"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Contact SBA
        </a>
      </li>
    </ul>
    <ul
      className="pageLinkGroup"
      id="tablet-footer-group-2"
    >
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-2-0"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA Team
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-2-1"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          FOIA
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-2-2"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Privacy Policy
        </a>
      </li>
    </ul>
    <ul
      className="pageLinkGroup"
      id="tablet-footer-group-3"
    >
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-3-0"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Newsroom
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-3-1"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Oversight & Advocacy
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="tablet-footer-group-3-2"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA en Espa√±ol
        </a>
      </li>
    </ul>
  </div>
  <div
    className="mobileFooterLinks"
  >
    <ul
      className="pageLinkGroup"
      id="mobile-footer-group-1"
    >
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-1-0"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          What We Do
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-1-1"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Newsroom
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-1-2"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          FOIA
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-1-3"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Contact SBA
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-1-4"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA en Espa√±ol
        </a>
      </li>
    </ul>
    <ul
      className="pageLinkGroup"
      id="mobile-footer-group-2"
    >
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-2-0"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA Team
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-2-1"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          SBA Performance
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-2-2"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Oversight & Advocacy
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-2-3"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Privacy Policy
        </a>
      </li>
      <li
        className="pageLink "
      >
        <a
          className="link "
          id="mobile-footer-group-2-4"
          onBlur={[Function]}
          onClick={[Function]}
          style={Object {}}
          target="_self"
        >
          Blog
        </a>
      </li>
    </ul>
  </div>
  <hr
    className="desktopRule"
  />
  <div
    className="subContainer"
  >
    <div
      className="SocialMediaLinkSet"
    >
      <span>
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          <img
            alt="SBA Facebook page"
            className="socialMediaIcons"
            src="test-file-stub"
          />
        </a>
      </span>
      <span>
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          <img
            alt="SBA Twitter page"
            className="socialMediaIcons"
            src="test-file-stub"
          />
        </a>
      </span>
      <span>
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          <img
            alt="SBA Youtube page"
            className="socialMediaIcons"
            src="test-file-stub"
          />
        </a>
      </span>
      <span>
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          <img
            alt="SBA Linked In page"
            className="socialMediaIcons"
            src="test-file-stub"
          />
        </a>
      </span>
      <span>
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          <img
            alt="SBA Instagram page"
            className="socialMediaIcons"
            src="test-file-stub"
          />
        </a>
      </span>
    </div>
    <form
      className="newsletter footer"
      data-testid="newsletter-form"
      onSubmit={[Function]}
    >
      <div>
        <p
          data-testid="newsletter-footer-title"
        >
          Sign up for SBA email updates
        </p>
        <div
          className="inputs"
        >
          <span
            className="buttonWrapper"
          >
            <a
              className="button alternate large link primary responsive spacing"
              data-testid="button"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              target="_self"
            >
              Subscribe
            </a>
          </span>
        </div>
      </div>
    </form>
    <hr
      className="mobileRule"
    />
    <div
      className="address"
    >
      <span
        className="name"
      >
        U.S. Small Business Administration
      </span>
      <span
        className="location"
      >
        409 3rd St, SW. Washington DC 20416
      </span>
    </div>
  </div>
</footer>
`;
